### PR TITLE
Add README for CLM PFT surface data

### PR DESCRIPTION
**Pull request by braghiere**
## Purpose 
Adds the CLM surface data file (at 0.9x1.25 degree resolution) with references for PFT distribution.

## Content
README.md

Review checklist

I have:
- followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
- followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
- followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
- checked that this PR does not duplicate an open PR.

In the Content, I have included 
- relevant unit tests, and integration tests, 
- appropriate docstrings on all functions, structs, and modules, and included relevant documentation.

----
- [X] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by Sbozzolo** [comment]
There are a couple of extra steps needed to be able to use this data (e.g., we need to assign IDs).

I'll add a checklist later and give you more concrete instruction.

**Comment by Sbozzolo** [comment]
The file `surfdata_0.9x1.25_hist_17pfts_nocft_CMIP6_simyr1700_c230809.nc` does not exist in `https://svn-ccsm-inputdata.cgd.ucar.edu/trunk/inputdata/lnd/clm2/surfdata_map/`

Could you please double check the links?

**Comment by Sbozzolo** [comment]
Got it! Is there going to be more `clm_data` in the future?

**Comment by braghiere** [comment]
Most likely yes. There are the PFTs look-up tables, e.g., Vcmax25(PFT), g1(PFT), etc. These will probably come from CLM too eventually.

**Comment by Sbozzolo** [comment]
> Most likely yes. There are the PFTs look-up tables, e.g., Vcmax25(PFT), g1(PFT), etc. These will probably come from CLM too eventually.

Okay, I marked the artifact as "undownloadable" (meaning that it is reponsability of the user to obtain it) and updated the relevant configuration on the cluster. Now we will be able to access this data with `@clima_artifact("clm_data")` in any of the packages where we add the entry to the `Artifacts.toml`.

Could you please update the your contribution making sure that everything is correct and updated and ensuring that the check boxes below are all checked? (I think it is mostly fixing the link, adding a link to the main README, and adding the text of the license)

Checklist:
- [X] I created a new folder `$artifact_name`
  - [ ] I added a `README.md` in that that folder that
    - [ ] describes the data and processing done to it
    - [ ] lists the sources of the raw data
    - [ ] lists the required citation, licenses
  - [ ] If applicable (e.g., for Creative Commons), I added a `LICENSE` file
  - [X] I added the scripts that retrieve, process, and produce the artifact
  - [X] I added the environment used for such scripts (typically, `Project.toml`
        and `Manifest.toml`)
  - [X] I added the `OutputArtifacts.toml` file containing the information
        needed for package developers to add `$artifact_name` to their package
- [X] I uploaded the artifact folder to the Caltech cluster (in
      `/groups/esm/ClimaArtifacts/artifacts/$artifact_name`)
- [X] I added the relevant code to the `Overides.toml` on the Caltech Cluster
      (in `/groups/esm/ClimaArtifacts/artifacts/Overrides.toml`)
- [ ] I added a link to the main `README.md` to point to the new artifact

Thank you!

File: clm_data/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Surface Data for CLM: Plant Functional Types
+
+## Overview
+This repository contains the `surfdata_0.9x1.25_16pfts__CMIP6_simyr2000_c170616.nc` file used in the Community Land Model (CLM) for historical climate modeling. The dataset is tailored for simulations that emphasize natural vegetation dynamics without the inclusion of cultivated crops (CFTs).
+
+## File Description
+The netCDF file includes comprehensive environmental data with a focus on vegetation represented through 17 different Plant Functional Types (PFTs). These PFTs play a crucial role in modeling biophysical processes and ecosystem functions within CLM simulations.
```
Comment (Sbozzolo) [inline clm_data/README.md:7]:
```suggestion
The netCDF file includes comprehensive environmental data with a focus on vegetation represented through 16 different Plant Functional Types (PFTs). These PFTs play a crucial role in modeling biophysical processes and ecosystem functions within CLM simulations.
```

**Review by Sbozzolo — APPROVED** [review]

**Pull request merged**